### ISSUE TITLE
about the result

### ISSUE THREAD

**Issue by SmallStriker** [shihao1895/SpatialActor#6]
I followed the steps mentioned in the project and replicated the experiment on a 3090 card. The loss function part showed normal results, but during the final test, the average success rate was only 30.22%. Could you please tell me if you have encountered this situation or if you have any suggestions？ Great thanks for any response.

<img width="1260" height="742" alt="Image" src="https://github.com/user-attachments/assets/c24ead71-c21e-4f6d-9f20-1b3b0a11b647" />

<img width="366" height="367" alt="Image" src="https://github.com/user-attachments/assets/9b0856c8-0070-4f13-a8b9-ca83023e1f5d" />

**Comment by shihao1895** [comment]
Hi,

One possible reason is the hardware setup. Our main experiments were run on 8 GPUs, while your reproduction used only a single 3090, so some performance gap is possible.

Recently, I also tested training with 2× RTX 4090, and the result on RLBench was 85.11, which is about 2 points lower than the full 8-GPU setting. So reducing compute does seem to cause some drop, but 30.22% is still much lower than expected.

I would also suggest checking whether there is any issue in the evaluation environment or evaluation pipeline, since bugs there can significantly affect the final success rate.

**Comment by SmallStriker** [comment]
Could you please tell me if the batch size (bs) remains 8 when using 2 instances of 4090 for training, and if gradient accumulation was employed to achieve an equivalent 8*8 batch size? Thank you very much for any of your responses.

**Comment by shihao1895** [comment]
The per-GPU batch size remains 8. No gradient accumulation was used.

**Comment by SmallStriker** [comment]
thanks very much

**Comment by SmallStriker** [comment]
May I ask what data collection framework you used in the real robot experiment and how you set it up?

**Comment by shihao1895** [comment]
For our real robot experiment, we based the data collection framework on RVT (`https://github.com/nvlabs/rvt`). We used a WidowX robotic arm and a Realsense D435 RGB-D camera. The data collection program was developed using ROS, capturing video at 30 fps, resizing it to 128x128, and manually annotating keyframes.

By the way, looking at it now, the RVT/PerAct codebase (we based it on) feels somewhat outdated. While it is suited for simpler tasks, it shows some limitations when applied to more complex tasks in real-world scenarios.
